subsys/nrf_security/src/drivers/cracen/cracenpsa/src/ikg_signature.c

```diff
@@ -63,9 +63,11 @@ static int exit_ikg(struct sx_pk_acq_req *pkreq)
 {
 
 	int status;
+	struct sx_pk_acq_req *oldreq = pkreq;
 
-	sx_pk_release_req(pkreq->req);
+	/* This locking may be recursive */
 	*pkreq = sx_pk_acquire_req(SX_PK_CMD_IK_EXIT);
+	sx_pk_release_req(oldreq);
```

Comment:
Uhm if you want to save the previous `pkreq` you need to copy its contents, not just make a pointer to it as its contents are replaced on the with what `sx_pk_acquire_req()` returns.
```suggestion
	struct sx_pk_acq_req oldreq = *pkreq;

	/* This locking may be recursive */
	*pkreq = sx_pk_acquire_req(SX_PK_CMD_IK_EXIT);
	sx_pk_release_req(&oldreq);
```

Reply:
Hah. Of course. Updated the PR